setup.py

```diff
@@ -327,13 +321,14 @@ def run(self):
 # to build with any sane version of Cython, so we should update this pin
 # periodically.
 #
-CYTHON_STR = 'Cython==0.29.14'
+CYTHON_STR = 'Cython==0.29.21'
 
 install_requires = [
     NUMPY_STR,
     'scipy >= 0.18.1',
     'smart_open >= 1.8.1',
     "dataclasses; python_version < '3.7'",  # pre-py3.7 needs `dataclasses` backport for use of `dataclass` in doc2vec.py
+    'six',
```

Comment:
What for? Looks like a regression, we already got rid of six in previous PRs IIRC.

Reply:
gensim wouldn't build without it, so I added it as a temporary workaround.

Upon further digging, we haven't removed it completely:

```
$ grep -R "from six" gensim
gensim/models/word2vec_corpusfile.pyx:from six import iteritems
gensim/models/word2vec_corpusfile.cpp: * from six import iteritems
gensim/models/word2vec_corpusfile.cpp: * from six import iteritems
```

plus the one occurrence I removed in this PR.

Reply:
Aha, I guess I only searched in `*.py` when I did my purging. I'll add a commit here for the `.pyx` and `.cpp` too, so that we're really six-free.

Reply:
I handled the pyx already.

The cpp is autogenerated, so we shouldn't touch it.